src/capi/object.cpp

```diff
@@ -146,7 +146,7 @@ extern "C" PyObject* PyObject_SelfIter(PyObject* obj) noexcept {
 extern "C" int PyObject_GenericSetAttr(PyObject* obj, PyObject* name, PyObject* value) noexcept {
     try {
         if (value == NULL)
-            delattrGeneric(obj, static_cast<BoxedString*>(name)->s, NULL);
```

Comment:
agreed.  we could also support llvm::Twine's, though those aren't as handy if callers aren't calling with concatenated strings.